_source/_posts/2023-01-23-keycloak-kubernetes-prod.md

```diff
@@ -0,0 +1,644 @@
+
+### Build the Spring Boot microservices architecture
+
+Install JHipster, you can do the classical local installation with npm.
+
+```bash
+npm install -g generator-jhipster@7.9.3
+```
+If you'd rather use Yarn or Docker, follow the instructions at [jhipster.tech](https://www.jhipster.tech/installation/#local-installation-with-npm-recommended-for-normal-users).
+
+Generate the microservices architecture from a JDL (JHipster-specific domain language) descriptor file. Create a folder for the project and add the file `reactive-ms.jdl` with the following content:
+
+```text
+application {
+  config {
+    baseName gateway
+    reactive true
+    packageName com.okta.developer.gateway
+    applicationType gateway
+    authenticationType oauth2
+    buildTool gradle
+    clientFramework vue
+    prodDatabaseType postgresql
+    serviceDiscoveryType consul
+    testFrameworks [cypress]
+  }
+  entities Blog, Post, Tag, Product
+}
+
+application {
+  config {
+    baseName blog
+    reactive true
+    packageName com.okta.developer.blog
+    applicationType microservice
+    authenticationType oauth2
+    buildTool gradle
+    databaseType neo4j
+    enableHibernateCache false
+    serverPort 8081
+    serviceDiscoveryType consul
+  }
+  entities Blog, Post, Tag
+}
+
+application {
+  config {
+    baseName store
+    reactive true
+    packageName com.okta.developer.store
+    applicationType microservice
+    authenticationType oauth2
+    buildTool gradle
+    databaseType mongodb
+    enableHibernateCache false
+    serverPort 8082
+    serviceDiscoveryType consul
+  }
+  entities Product
+}
+
+entity Blog {
+  name String required minlength(3)
+  handle String required minlength(2)
+}
+
+entity Post {
+  title String required
+  content TextBlob required
+  date Instant required
+}
+
+entity Tag {
+  name String required minlength(2)
+}
+
+entity Product {
+  title String required
+  price BigDecimal required min(0)
+  image ImageBlob
+}
+
+relationship ManyToOne {
+  Blog{user(login)} to User with builtInEntity
+  Post{blog(name)} to Blog
+}
+
+relationship ManyToMany {
+  Post{tag(name)} to Tag{post}
+}
+
+paginate Post, Tag with infinite-scroll
+paginate Product with pagination
+
+microservice Product with store
+microservice Blog, Post, Tag with blog
+
+deployment {
+  deploymentType docker-compose
+  appsFolders [gateway, blog, store]
+  dockerRepositoryName "indiepopart"
+}
+```
+
+Repalce the `dockerRepositoryName` with your account name. Then run the following command:
+
+```shell
+jhipster jdl reactive-ms.jdl
+```
+
+After the generation, you will find sub-folders were created for the `gateway`, `store`, and `blog` services. The `gateway` will act as the front-end application and as a secure router to the `store` and `blog` microservices.
+
+Create a [Docker Hub](https://hub.docker.com/) personal account, if you don't have one. Then build and publish each application image. For example, in the `gateway` folder run:
```

Comment:
This might need to be moved up (or mentioned in prerequisites) because it's used in the JDL before this part.

Reply:
Moved up and added to required services